nservicebus/messaging/routing-extensibility_content_core_[6,).partial.md

```diff
@@ -0,0 +1,61 @@
+
+The routing system prevents route ambiguity. If the new or replaced routes conflict with existing ones, an exception is thrown. It is up to the route extension to deal with that exception but usually it is best practice to trigger the endpoint shutdown to prevent misrouting of messages.
+
+snippet:RoutingExtensibility-TriggerEndpointShutdown 
+
+## Event routing
+
+As mentioned in the [general routing documentation](/nservicebus/messaging/routing.md), event routing differs depending on the transport capabilities. 
+
+Transports supporting the [Publish-Subscribe](/nservicebus/messaging/publish-subscribe/) pattern natively ({include: transports-multicast}) implement the event routing themselves. Refer to specific transport documentation for details on extensibility points.
+
+Transports without that support ({include: transports-unicast}) relay on NServiceBus core routing for event delivery. To emulate multicast routing on top of a unicast transport NServiceBus uses. The key concept is the collection of publishers. For each event it contains information on the logical endpoint that publishes it. Routing extensions can access the publishers collections from the `Feature` set up code:
+
+snippet:RoutingExtensibility-Publishers
+
+The source parameter has the same meaning and effect as in the routes collection.
+
+The publishers collection behaves the same way as the route table with regards to threading and atomicity.
+
+## Physical routing
+
+Physical routing is responsible for mapping the destination logical endpoint to the actual transport address (queue name). 
+
+When using a broker transport ({include:transports-broker}), the physical routing is entirely managed by NServiceBus and does not require any configuration.
+
+When using a bus transport ({include:transports-bus}), the physical routing is important because the transport address has to contain the information about the node of the bus that the endpoint is using. In MSMQ each machine runs a single node of the MSMQ system.
+
+Routing extensions can influence the physical routing by modifying the endpoint instances collection:
+
+snippet:RoutingExtensibility-Instances
+
+The source parameter has the same meaning and effect as in the routes collection.
+
+The instances collection is thread-safe. It allows registering multiple instance of a given endpoint. In case there is more than one, message distribution is involved. 
+
+## Message distribution
+
+When there are multiple instances of a single logical endpoint registered in the routing system, each outgoing message has to undergo the distribution process to determine which instance is going to receive this particular message. All NServiceBus messages are always delivered to a single physical instance of any given logical receiver. By default a round-robin algorithm is used to determine the destination. Routing extensions can override this behavior by registering a custom `DistributionStrategy` for a given destination endpoint.
```

Comment:
> When there are multiple instances of a single logical endpoint registered in the routing system, each outgoing message has to undergo the distribution process to determine which instance is going to receive this particular message. All NServiceBus messages are always delivered to a single physical instance of any given logical receiver. 

Every message is always delivered to a single physical instance of the logical endpoint. When the endpoint is scaled-out (i.e. there are multiple instances of a single logical endpoint), then NServiceBus by default determines the destination for a message in a round-robin fashion.